service/interpreter/continueAsNewer.go

```diff
@@ -110,27 +113,56 @@ func LoadInternalsFromPreviousRun(
 	return &resp, nil
 }
 
+func (c *ContinueAsNewer) GetSnapshot() service.ContinueAsNewDumpResponse {
+	localStateExecutionToResumeMap := map[string]service.StateExecutionResumeInfo{}
+	for key, state := range c.StateExecutionToResumeMap {
+		localStateExecutionToResumeMap[key] = state
+	}
+	for _, value := range c.stateRequestQueue.GetAllStateResumeRequests() {
+		localStateExecutionToResumeMap[value.StateExecutionId] = value
+	}
+	return service.ContinueAsNewDumpResponse{
+		InterStateChannelReceived:  c.interStateChannel.GetAllReceived(),
+		SignalsReceived:            c.signalReceiver.GetAllReceived(),
+		StateExecutionCounterInfo:  c.stateExecutionCounter.Dump(),
+		DataObjects:                c.persistenceManager.GetAllDataObjects(),
+		SearchAttributes:           c.persistenceManager.GetAllSearchAttributes(),
+		StatesToStartFromBeginning: c.stateRequestQueue.GetAllStateStartRequests(),
+		StateExecutionsToResume:    localStateExecutionToResumeMap,
+		StateOutputs:               c.outputCollector.GetAll(),
+		StaleSkipTimerSignals:      c.timerProcessor.Dump(),
+	}
+}
+
 func (c *ContinueAsNewer) SetQueryHandlersForContinueAsNew(ctx UnifiedContext) error {
-	return c.provider.SetQueryHandler(ctx, service.ContinueAsNewDumpQueryType, func() (*service.ContinueAsNewDumpResponse, error) {
-		localStateExecutionToResumeMap := map[string]service.StateExecutionResumeInfo{}
-		for key, state := range c.StateExecutionToResumeMap {
-			localStateExecutionToResumeMap[key] = state
-		}
-		for _, value := range c.stateRequestQueue.GetAllStateResumeRequests() {
-			localStateExecutionToResumeMap[value.StateExecutionId] = value
-		}
-		return &service.ContinueAsNewDumpResponse{
-			InterStateChannelReceived:  c.interStateChannel.GetAllReceived(),
-			SignalsReceived:            c.signalReceiver.GetAllReceived(),
-			StateExecutionCounterInfo:  c.stateExecutionCounter.Dump(),
-			DataObjects:                c.persistenceManager.GetAllDataObjects(),
-			SearchAttributes:           c.persistenceManager.GetAllSearchAttributes(),
-			StatesToStartFromBeginning: c.stateRequestQueue.GetAllStateStartRequests(),
-			StateExecutionsToResume:    localStateExecutionToResumeMap,
-			StateOutputs:               c.outputCollector.GetAll(),
-			StaleSkipTimerSignals:      c.timerProcessor.Dump(),
-		}, nil
-	})
+	return c.provider.SetQueryHandler(ctx, service.ContinueAsNewDumpQueryType,
```

Comment:
I'm having a hard time understanding how/where this pagination code (SetQueryHandler) is called when the workflow is queried. What is the code path?

Reply:
better to use your own IDE to browse the code path:
* workflowImpl.go: `if input.IsResumeFromContinueAsNew { LoadInternalsFromPreviousRun(...) }`
* continueAsNewer.go: `provider.ExecuteActivity(&resp, false, ctx, DumpWorkflowInternal...`
* activityImpl.go: `apiClient.DefaultApi.ApiV1WorkflowInternalDumpPost(ctx)`
* service.go: `s.client.QueryWorkflow(ctx, &pageOfSnapshot, request.GetWorkflowId(), request.GetWorkflowRunId(), service.ContinueAsNewDumpQueryType, request)`
* continueAsNewer.go : the registered handler of `SetQueryHandler(ctx, service.ContinueAsNewDumpQueryType,`